device/pcie/pci_device.cpp

```diff
@@ -748,19 +756,6 @@ bool PCIDevice::init_hugepage(uint32_t num_host_mem_channels) {
             continue;
         }
 
-        // Beter performance if hugepage just allocated (populate flag to prevent lazy alloc) is migrated to same
-        // numanode as TT device.
-        if (!tt::cpuset::tt_cpuset_allocator::bind_area_to_memory_nodeset(physical_device_id, mapping, hugepage_size)) {
```

Comment:
So the above numa_set_preferred should offer similar/same functionality as this old code?

Reply:
Yes, the main behavioral difference is the code will now attempt to allocate the page on the desired node (rather than allocate it and then attempt to migrate it).